mise/tasks/vm/create-gcp

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+#MISE description="Create a development VM within GCP for a project"
+#USAGE arg "<project_name>" help="Short name of the development project (for example, 'one' or '$feature-name')"
+#USAGE flag "--project <project>" help="GCP project ID" default="estuary-theatre"
+#USAGE flag "--zone <zone>" help="Zone in which to create the VM" default="us-central1-a"
+#USAGE flag "--machine-type <machine_type>" help="GCP machine type" default="c4-highcpu-8"
+#USAGE flag "--image-family <image_family>" help="GCP machine image" default="ubuntu-2404-lts-amd64"
+
+ACCOUNT=$(gcloud auth list --filter=status:ACTIVE --format="value(account)")
+USERNAME="${ACCOUNT%%@*}"
+
+# Derive a deterministic instance name: dev-<user>-<project-name>
+INSTANCE_NAME="dev-${USERNAME}-${usage_project_name:?}"
+
+# Validate ssh-agent is running and has keys
+if ! ssh-add -L &>/dev/null; then
+  echo "ERROR: No SSH keys found in ssh-agent." >&2
+  echo "Please run 'ssh-add' to add your keys before creating a VM." >&2
+  echo "" >&2
+  echo "Example: ssh-add ~/.ssh/id_ed25519" >&2
+  exit 1
+fi
+
+# Extract public keys from ssh-agent for instance metadata
+# Format: username:ssh-key-type key-data comment
+SSH_KEYS=$(ssh-add -L | sed "s/^/${USERNAME}:/")
+
+echo "Using SSH keys from agent:"
+ssh-add -l | sed 's/^/  - /'
+echo ""
+
+gcloud compute instances create ${INSTANCE_NAME} \
+  --project ${usage_project:?} \
+  --zone ${usage_zone:?} \
+  --machine-type ${usage_machine_type:?} \
+  --image-family ${usage_image_family:?} \
+  --image-project ubuntu-os-cloud \
+  --network-interface subnet=default \
+  --metadata ssh-keys="${SSH_KEYS}" \
+  --tags dev-vm \
+  --labels dev-vm=true,owner=${USERNAME} \
+  --boot-disk-size 128GB \
+  --boot-disk-type hyperdisk-balanced \
+
+# Add additional .bashrc configuration if not already present.
+if ! grep -q gcp-vms ~/.ssh/config; then
+    cat >> ~/.ssh/config << 'EOF'
```

Comment:
Note: Appending this include directive doesn't work right if you already have an SSH config with Host or Match directives, ideally it should be at the top of the file.